File: sample/src/main/java/jesusm/com/fingerprintmanager/sample/MainActivity.java
package jesusm.com.fingerprintmanager.sample;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.StyleRes;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.jesusm.jfingerprintmanager.JFingerprintManager;

public class MainActivity extends AppCompatActivity {

    private static final String KEY = "my_key";
    private TextView messageTextView;
    private EditText messageToBeEncryptedEditText;
    private Button authenticateButton;
    private Button encryptTextButton;
    private Button decryptTextButton;

    private String messageToDecrypt;
    @StyleRes
    private int dialogTheme;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        initClickListeners();
    }

    private void initViews() {
        messageTextView = (TextView) findViewById(R.id.message);
        messageToBeEncryptedEditText = (EditText) findViewById(R.id.editText);
        authenticateButton = (Button) findViewById(R.id.buttonAuthenticate);
        encryptTextButton = (Button) findViewById(R.id.buttonEncrypt);
        decryptTextButton = (Button) findViewById(R.id.buttonDecrypt);

        selectView(findViewById(R.id.buttonDialogThemeLight));
    }

    private JFingerprintManager createFingerprintManagerInstance() {
        JFingerprintManager fingerprintManager = new JFingerprintManager(this, KEY);
        fingerprintManager.setAuthenticationDialogStyle(dialogTheme);
        return fingerprintManager;
    }

    private void initClickListeners() {

        findViewById(R.id.buttonDialogThemeLight).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                selectView(v);

                deselectView(findViewById(R.id.buttonDialogThemeDark));

                dialogTheme = R.style.DialogThemeLight;
            }
        });

        findViewById(R.id.buttonDialogThemeDark).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                selectView(v);

                deselectView(findViewById(R.id.buttonDialogThemeLight));

                dialogTheme = R.style.DialogThemeDark;
            }
        });

        authenticateButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                createFingerprintManagerInstance().startAuthentication(new JFingerprintManager.AuthenticationCallback() {
                    @Override
                    public void onAuthenticationSuccess() {
                        messageTextView.setText("Successfully authenticated");
                    }

                    @Override
                    public void onSuccessWithManualPassword(@NonNull String password) {
                        messageTextView.setText("Manual password: " + password);
                    }

                    @Override
                    public void onFingerprintNotRecognized() {
                        messageTextView.setText("Fingerprint not recognized");
                    }

                    @Override
                    public void onAuthenticationFailedWithHelp(String help) {
                        messageTextView.setText(help);
                    }

                    @Override
                    public void onFingerprintNotAvailable() {
                        messageTextView.setText("Fingerprint not available");
                    }

                    @Override
                    public void onCancelled() {
                        messageTextView.setText("Operation cancelled by user");
                    }
                }, getSupportFragmentManager());
            }
        });

        encryptTextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                messageToDecrypt = messageToBeEncryptedEditText.getText().toString();
                createFingerprintManagerInstance().encrypt(messageToDecrypt, new JFingerprintManager.EncryptionCallback() {
                    @Override
                    public void onFingerprintNotRecognized() {
                        messageTextView.setText("Fingerprint not recognized");
                    }

                    @Override
                    public void onAuthenticationFailedWithHelp(String help) {
                        messageTextView.setText(help);
                    }

                    @Override
                    public void onFingerprintNotAvailable() {
                        messageTextView.setText("Fingerprint not available");
                    }

                    @Override
                    public void onEncryptionSuccess(String messageEncrypted) {
                        String message = getString(R.string.encrypt_message_success, messageEncrypted);
                        messageTextView.setText(message);
                        messageToBeEncryptedEditText.setText(messageEncrypted);
                        encryptTextButton.setVisibility(View.GONE);
                        decryptTextButton.setVisibility(View.VISIBLE);
                    }

                    @Override
                    public void onEncryptionFailed() {
                        messageTextView.setText("Encryption failed");
                    }

                    @Override
                    public void onCancelled() {
                        messageTextView.setText("Operation cancelled by user");
                    }

                }, getSupportFragmentManager());
            }
        });

        decryptTextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                messageToDecrypt = messageToBeEncryptedEditText.getText().toString();
                createFingerprintManagerInstance().decrypt(messageToDecrypt, new JFingerprintManager.DecryptionCallback() {
                    @Override
                    public void onDecryptionSuccess(String messageDecrypted) {
                        String message = getString(R.string.decrypt_message_success, messageDecrypted);
                        messageTextView.setText(message);
                        messageToBeEncryptedEditText.setText("");
                        decryptTextButton.setVisibility(View.GONE);
                        encryptTextButton.setVisibility(View.VISIBLE);
                    }

                    @Override
                    public void onDecryptionFailed() {
                        messageTextView.setText("Decryption failed");
                    }

                    @Override
                    public void onFingerprintNotRecognized() {
                        messageTextView.setText("Fingerprint not recognized");
                    }

                    @Override
                    public void onAuthenticationFailedWithHelp(String help) {
                        messageTextView.setText(help);
                    }

                    @Override
                    public void onFingerprintNotAvailable() {
                        messageTextView.setText("Fingerprint not available");
                    }

                    @Override
                    public void onCancelled() {
                        messageTextView.setText("Operation cancelled by user");
                    }
                }, getSupportFragmentManager());
            }
        });
    }

    private void selectView(View view) {
        view.setSelected(true);
        view.setElevation(32);
    }

    private void deselectView(View view) {
        view.setSelected(false);
        view.setElevation(0);
    }
}

File: sample/src/main/java/jesusm/com/fingerprintmanager/sample/MainActivity.kt
package jesusm.com.fingerprintmanager.sample

import android.os.Bundle
import android.support.annotation.StyleRes
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import com.jesusm.kfingerprintmanager.KFingerprintManager
import com.jesusm.kfingerprintmanager.utils.bind

class MainActivity : AppCompatActivity() {
    private val KEY = "my_key"

    private var messageTextView = bind<TextView>(R.id.message)
    private var messageToBeEncryptedEditText = bind<EditText>(R.id.editText)
    private var authenticateButton = bind<Button>(R.id.buttonAuthenticate)
    private var encryptTextButton = bind<Button>(R.id.buttonEncrypt)
    private var decryptTextButton = bind<Button>(R.id.buttonDecrypt)
    @StyleRes
    private var dialogTheme: Int = 0

    private var messageToDecrypt: String = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        selectView(findViewById(R.id.buttonDialogThemeLight))

        initClickListeners()
    }

    private fun initClickListeners() {

        findViewById<Button>(R.id.buttonDialogThemeLight).setOnClickListener { v ->
            selectView(v)

            deselectView(findViewById(R.id.buttonDialogThemeDark))

            dialogTheme = R.style.DialogThemeLight
        }

        findViewById<Button>(R.id.buttonDialogThemeDark).setOnClickListener { v ->
            selectView(v)

            deselectView(findViewById(R.id.buttonDialogThemeLight))

            dialogTheme = R.style.DialogThemeDark
        }

        authenticateButton.value.setOnClickListener {
            createFingerprintManagerInstance().authenticate(object : KFingerprintManager.AuthenticationCallback {
                override fun onAuthenticationFailedWithHelp(help: String?) {
                    messageTextView.value.text = help
                }

                override fun onAuthenticationSuccess() {
                    messageTextView.value.text = "Successfully authenticated"
                }

                override fun onSuccessWithManualPassword(password: String) {
                    messageTextView.value.text = "Manual password: " + password
                }

                override fun onFingerprintNotRecognized() {
                    messageTextView.value.text = "Fingerprint not recognized"
                }

                override fun onFingerprintNotAvailable() {
                    messageTextView.value.text = "Fingerprint not available"
                }

                override fun onCancelled() {
                    messageTextView.value.text = "Operation cancelled by user"
                }
            }, supportFragmentManager)
        }

        encryptTextButton.value.setOnClickListener {
            messageToDecrypt = messageToBeEncryptedEditText.value.text.toString()
            createFingerprintManagerInstance().encrypt(messageToDecrypt, object : KFingerprintManager.EncryptionCallback {
                override fun onFingerprintNotRecognized() {
                    messageTextView.value.text = "Fingerprint not recognized"
                }

                override fun onAuthenticationFailedWithHelp(help: String?) {
                    messageTextView.value.text = help
                }

                override fun onFingerprintNotAvailable() {
                    messageTextView.value.text = "Fingerprint not available"
                }

                override fun onEncryptionSuccess(messageEncrypted: String) {
                    val message = getString(R.string.encrypt_message_success, messageEncrypted)
                    messageTextView.value.text = message
                    messageToBeEncryptedEditText.value.setText(messageEncrypted)
                    encryptTextButton.value.visibility = View.GONE
                    decryptTextButton.value.visibility = View.VISIBLE
                }

                override fun onEncryptionFailed() {
                    messageTextView.value.text = "Encryption failed"
                }

                override fun onCancelled() {
                    messageTextView.value.text = "Operation cancelled by user"
                }
            }, supportFragmentManager)
        }

        decryptTextButton.value.setOnClickListener {
            messageToDecrypt = messageToBeEncryptedEditText.value.text.toString()
            createFingerprintManagerInstance().decrypt(messageToDecrypt, object : KFingerprintManager.DecryptionCallback {
                override fun onDecryptionSuccess(messageDecrypted: String) {
                    val message = getString(R.string.decrypt_message_success, messageDecrypted)
                    messageTextView.value.text = message
                    messageToBeEncryptedEditText.value.setText("")
                    decryptTextButton.value.visibility = View.GONE
                    encryptTextButton.value.visibility = View.VISIBLE
                }

                override fun onDecryptionFailed() {
                    messageTextView.value.text = "Decryption failed"
                }

                override fun onFingerprintNotRecognized() {
                    messageTextView.value.text = "Fingerprint not recognized"
                }

                override fun onAuthenticationFailedWithHelp(help: String?) {
                    messageTextView.value.text = help
                }

                override fun onFingerprintNotAvailable() {
                    messageTextView.value.text = "Fingerprint not available"
                }

                override fun onCancelled() {
                    messageTextView.value.text = "Operation cancelled by user"
                }
            }, supportFragmentManager)
        }
    }

    private fun createFingerprintManagerInstance(): KFingerprintManager {
        val fingerprintManager = KFingerprintManager(this, KEY)
        fingerprintManager.setAuthenticationDialogStyle(dialogTheme)
        return fingerprintManager
    }

    private fun selectView(view: View) {
        view.apply {
            isSelected = true
            elevation = 32f
        }
    }

    private fun deselectView(view: View) {
        view.apply {
            isSelected = false
            elevation = 0f
        }
    }
}
